src/language/concurrency.md

```diff
@@ -237,303 +287,272 @@ After handling the events, the isolate exits.
-the send fails if any transitively reachable object is unsendable.
-For example, you can send an object of type `List<Object>` only if
-none of the objects in the list is unsendable.
-If one of the objects is, say, a `Socket`, then
-the send fails because sockets are unsendable.
-
-For information on the kinds of objects that you can send in messages,
-see the API reference documentation for the [`send()` method][].
-
 A worker isolate can perform I/O
-(reading and writing files, for example), set timers, and more.
-It has its own memory and
-doesn't share any state with the main isolate.
-The worker isolate can block without affecting other isolates.
-
-[`send()` method]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/SendPort/send.html
-
-## Code examples
-
-This section discusses some examples
-that use the `Isolate` API
-to implement isolates.
-
-### Implementing a simple worker isolate
-
-These examples implement a main isolate
-that spawns a simple worker isolate.
-[`Isolate.run()`][] simplifies the steps behind
-setting up and managing worker isolates:
-
-1. Spawns (starts and creates) an isolate
-2. Runs a function on the spawned isolate
-3. Captures the result
-4. Returns the result to the main isolate
-5. Terminates the isolate once work is complete
-6. Checks, captures, and throws exceptions and errors back to the main isolate
-
-[`Isolate.run()`]: {{site.dart-api}}/dev/dart-isolate/Isolate/run.html
-
-{{site.alert.flutter-note}}
-  If you're using Flutter,
-  you can use [Flutter's `compute` function][]
-  instead of `Isolate.run()`.
-  On the [web](#web), the `compute` function falls back
-  to running the specified function on the current event loop.
-  Use `Isolate.run()` when targeting native platforms only,
-  for a more ergonomic API.
-{{site.alert.end}}
-
-[native and non-native platforms]: /overview#platform
-[Flutter's `compute` function]: {{site.flutter-api}}/flutter/foundation/compute.html
+(reading and writing files, for example), set timers, and more. It has its own
+memory and doesn't share any state with the main isolate. The worker isolate can
+block without affecting other isolates.
 
-#### Running an existing method in a new isolate
+### Isolate syntax
 
-The main isolate contains the code that spawns a new isolate: 
+There are two ways to work with isolates in Dart, depending on the use-case: 
 
-<?code-excerpt "lib/simple_worker_isolate.dart (main)"?>
-```dart
-const String filename = 'with_keys.json';
+* Use [`Isolate.run()`][] to perform a single computation on a separate thread.
+* Use [`Isolate.spawn()`][] to create an isolate that will handle
+  multiple messages over time, or a background worker.
 
-void main() async {
-  // Read some data.
-  final jsonData = await Isolate.run(_readAndParseJson);
+In most cases, `Isolate.run` is the recommended 
+API to run processes in the background.
 
-  // Use that data.
-  print('Number of JSON keys: ${jsonData.length}');
-}
-```
+### `Isolate.run()`
```

Comment:
Is this intended to be a subheading of the previous section similar to the `Isolate.spawn` section below?

```suggestion
#### `Isolate.run()`
```